src/main/java/gyro/google/compute/RegionDiskResource.java

```diff
@@ -0,0 +1,268 @@
+            : null;
+    }
+
+    @Override
+    public void setResourcePolicies(List<String> resourcePolicies) {
+        super.setResourcePolicies(resourcePolicies != null
+            ? resourcePolicies.stream().map(p -> ComputeUtils.toResourcePolicyUrl(getProjectId(), p, getRegion())).collect(Collectors.toList())
+            : null);
+    }
+
+    @Override
+    public void setType(String type) {
+        super.setType(type != null ? ComputeUtils.toRegionDiskTypeUrl(getProjectId(), type, getRegion()) : null);
+    }
+
+    @Override
+    public void copyMore(Disk disk) {
+        setRegion(disk.getRegion().substring(disk.getRegion().lastIndexOf("/") + 1));
+        setReplicaZones(disk.getReplicaZones());
+        setType(disk.getType());
+        setResourcePolicies(disk.getResourcePolicies());
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state, Disk disk) {
+        Compute client = createComputeClient();
+
+        disk.setRegion(getRegion());
+        disk.setReplicaZones(getReplicaZones());
+
+        try {
+            Compute.RegionDisks.Insert insert = client.regionDisks().insert(getProjectId(), getRegion(), disk);
+            Operation operation = insert.execute();
+            Operation.Error error = waitForCompletion(client, operation);
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+
+            refresh();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public List<ValidationError> validateMore() {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getReplicaZones().size() != 2) {
+            errors.add(new ValidationError(
+                this,
+                "replica-zones",
+                "Disk requires exactly two replica zones."));
+        }
+
+        return errors;
+    }
+
+    @Override
+    public boolean refresh() {
+        Compute client = createComputeClient();
+
+        try {
+            Disk disk = client.regionDisks().get(getProjectId(), getRegion(), getName()).execute();
+            copyFrom(disk);
+
+            return true;
+        } catch (GoogleJsonResponseException je) {
+            if (je.getDetails().getMessage().matches("The resource (.*) was not found")) {
+                return false;
+            } else {
+                throw new GyroException(je.getDetails().getMessage());
+            }
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) {
+        Compute compute = createComputeClient();
+
+        try {
+            compute.regionDisks().delete(getProjectId(), getRegion(), getName()).execute();
+        } catch (IOException ex) {
```

Comment:
Should catch the `GoogleJsonResponseException`, which has better details of the error message.
```suggestion
        } catch (GoogleJsonResponseException ex) {
```